### PR TITLE
Fix the install of snapshot CRDs and controller

### PR DESCRIPTION
**Pull request by xing-yang**
This PR installs snapshot CRDs and rbac rules from the repo
and installs snapshot controller from a local image if it is
a PR in external-snapshotter repo.
Otherwise it uses main or a stable version.

### CODE REVIEW

File: prow.sh [hunk 1]
```diff
@@ -733,8 +746,55 @@ install_snapshot_controller() {
     sleep 10   
   done
 
+  SNAPSHOT_CONTROLLER_YAML="${CONTROLLER_DIR}/deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml"
+  if [[ ${REPO_DIR} == *"external-snapshotter"* ]]; then
+      NEW_TAG="csiprow"
```
Comment (msau42) [inline prow.sh:751]: Can you add a comment where you got this logic from?

**Comment by xing-yang** [comment]
Submitted the following PRs to test this fix.  CI has passed on both PRs:
https://github.com/kubernetes-csi/external-snapshotter/pull/427
https://github.com/kubernetes-csi/csi-driver-host-path/pull/224

File: prow.sh [hunk 2]
```diff
@@ -733,8 +746,55 @@ install_snapshot_controller() {
...
+      # shellcheck disable=SC2086
+      # Ignore: Use find instead of ls to better handle non-alphanumeric filenames.
+      # shellcheck disable=SC2012
+      for i in $(ls ${SNAPSHOT_CONTROLLER_YAML} | sort); do
+          echo "   $i"
+          # Ignore: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
+          # shellcheck disable=SC2002
+          # Ignore: See if you can use ${variable//search/replace} instead.
+          # shellcheck disable=SC2001
+          modified="$(cat "$i" | while IFS= read -r line; do
```
Comment (msau42) [inline prow.sh:768]: I think longer term we should consider using kustomize to patch the image so we don't need to maintain this logic.

Comment (msau42) [reply prow.sh:768]: https://github.com/kubernetes-csi/csi-release-tools/issues/114

File: prow.sh (hunk 1 above)
Comment (xing-yang) [reply prow.sh:751]: This is here: https://github.com/kubernetes-csi/csi-release-tools/pull/113/files#diff-c2d96beb8cf73c62f7ae059aff96e381a8d73317df3ce356f36ba6b46338959dR1115

The image built from PR gets a "csiprow" tag.

File: prow.sh (hunk 2 above)
Comment (xing-yang) [reply prow.sh:768]: Sure

File: prow.sh (hunk 1 above)
Comment (xing-yang) [reply prow.sh:753]: Added comments.

**Comment by xing-yang** [comment]
@msau42 addressed your comments.  PTAL.  Thanks.

File: prow.sh (hunk 1 above)
Comment (msau42) [reply prow.sh:753]: Oh sorry, I was referring to the logic below where we're replacing the image tag

Comment (xing-yang) [reply prow.sh:753]: That's from deploy-hostpath.sh: https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/deploy/util/deploy-hostpath.sh#L155

Comment (xing-yang) [reply prow.sh:753]: Updated.

**Pull request merged**